### ISSUE TITLE
Vulkan: qwen3next MoE with IQ1_M router weights produces all-zero/garbage output (works on CPU)

### ISSUE THREAD

**Issue by 21307369** [ggml-org/llama.cpp#27022]
## Environment
- macOS 23.6 (x86_64 Hackintosh), MoltenVK 1.4.2, AMD RX 6750 GRE 12GB
- llama.cpp b10310 (commit 3653e6d), Vulkan backend, self-built with AppleClang 16

## Model
- `nerkyor/Qwen3.6-35B-A3B-DSV4Pro-SFT-GPT56Sol-RL-Agent-GGUF` Q4 LynnStyle (20.47GB, qwen3next arch, 40 layers, 256 experts top-8, SSM/recurrent layers)
- Tensor type distribution: F32 261 (norms, ssm_a/conv1d/dt), Q8_1 36 (attention), Q4_K 45 + Q5_K 30 + Q6_K 222 + Q8_K 39 (experts/SSM weights), **IQ1_M 40 = all `ffn_gate_inp_shexp.weight` (MoE router)**

## Reproduction
```bash
llama-server -m Qwen3.6-35B-A3B-DSV4Pro-SFT-GPT56Sol-RL-Agent-LynnStyle-Q4-imatrix.gguf \
  -ngl 99 --n-cpu-moe 30 -t 5 -c 8192 --host 127.0.0.1 --port 8080
```
Then chat completion with a **multi-line prompt** (e.g. JSON compression instruction with newlines). Output is 300× `0` characters (or garbage tokens), temperature 0.1.

Single-line prompts work fine. Full-CPU (`-ngl 0`) works fine for all prompts.

## Observed
1. Multi-line prompts → all-zero output (`000000...`) or garbage tokens on Vulkan offload
2. After a failed request, **subsequent requests stay corrupted until server restart**
3. Overriding the IQ1_M router + SSM tensors to CPU fixes output quality:
   `--override-tensor "blk\.[0-9]+\.ffn_gate_inp_shexp\.weight=CPU" --override-tensor "blk\.[0-9]+\.ssm_.*=CPU"` → correct output, but ~2 tok/s (too slow)
4. A control model with identical architecture but F32 router weights (`Qwen3.6-35B-A3B-UD-Q4_K_M.gguf`, qwen35moe arch) works perfectly on Vulkan with same offload settings

## Expected
Correct output on Vulkan backend regardless of prompt formatting

## Analysis
Vulkan backend has full IQ1_M support (dequant_iq1_m.comp, mul_mat_vec_iq1_m.comp, matmul_iq1_m_f16/f32). The failure is structural (all-zero logits → all-zero tokens), not minor precision loss. CPU path produces correct results with the same weights, pointing to a numerical issue in the Vulkan IQ1_M dequant/matmul path (possibly triggered by specific activation patterns from newline tokens in the router input).

## Additional
- Failure state persists across requests until server restart (state pollution)
- Not reproducible on full CPU offload
- Control: UD model (same 35B-A3B MoE + SSM architecture, F32 router) has zero issues on Vulkan
